packages/component-library/src/components/plmg-radio-button-group/plmg-radio-button-group.tsx

```diff
@@ -0,0 +1,144 @@
+      throw new Error('name must be a string');
+  }
+
+  /**
+   * Define text for the form's label
+   */
+  @Prop() label: string;
+  @Watch('label')
+  validateLabel(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('label: required');
+    if (newValue && typeof newValue !== 'string')
+      throw new Error('label must be a string');
+  }
+
+  /**
+   * Define size of all radio button's in radio button group.
+   *
+   * Allowed values:
+   *   - medium
+   *   - large
+   *
+   * Default: medium
+   */
+  @Prop() size: PlmgRadioButtonSize = 'medium';
+  @Watch('size')
+  validateSize(newValue: string) {
+    if (!isPlmgRadioButtonSize(newValue))
+      throw new Error('size: must be a valid value');
+  }
+
+  /**
+   * Define radio group's required status
+   *
+   * Allowed values:
+   *   - true
+   *   - false
+   *
+   * Default: false
+   */
+  @Prop() required: boolean = false;
+  @Watch('required')
+  validatesRequired(newValue: boolean) {
+    if (typeof newValue !== 'boolean')
+      throw new Error('required: must be boolean');
+  }
+
+  /**
+   * Define each radio button's value
+   *
+   * Accepts an array or JSON string
+   */
+  @Prop() values: string[] | string;
+  @State() parsedValues: string[];
+
+  componentWillLoad() {
+    this.parseValuesProp(this.values);
+  }
+
+  @Watch('values')
+  parseValuesProp(newValue: string | string[]) {
+    if (newValue) {
+      this.parsedValues = Array.isArray(newValue)
+        ? newValue
+        : JSON.parse(newValue);
+    }
+  }
+  validatesValues(newValue: string[] | string) {
+    if (
+      typeof newValue !== 'object' ||
+      (typeof newValue === 'object' &&
+        !Array.isArray(newValue) &&
+        typeof newValue !== 'string')
+    ) {
+      throw new Error('values: must be array or string');
+    }
+    if (
+      Array.isArray(newValue) &&
+      newValue.some((value) => typeof value !== 'string')
+    ) {
+      throw new Error('values: values within array must be strings');
+    }
+  }
+
+  /**
+   * Define error message for radio group
+   *
+   * Will render one error message for the
+   * radio button group, affects styling of
+   * all radio buttons in group
+   */
+  @Prop() errorMessage?: string | undefined;
+  @Watch('errorMessage')
+  validatesErrorMessage(newValue: string) {
+    if (newValue !== undefined && typeof newValue !== 'string')
+      throw new Error('errorMessage: must be a string if provided');
+  }
+
+  render() {
+    const legendClasses = {
+      large: this.size === 'large',
+    };
+
+    return (
+      <fieldset>
+        <legend class={legendClasses}>
+          {this.label}
+          {this.required && <span>*</span>}
+        </legend>
+        {this.parsedValues.map((radio) => (
+          <plmg-radio-button
+            value={radio}
+            name={this.name}
+            size={this.size}
+            highlighted={!!this.errorMessage}
```

Comment:
Need to pass down `required` to make sure the form cannot be submitted when nothing is selected.
https://www.w3docs.com/snippets/html/how-to-use-the-required-attribute-with-the-radio-input-field.html
At least one of the radio button must have the required attribute.

It will enable the default browser behavior but won't display the nice error message :shrug: Maybe see with designers?